common/client-core/src/client/base_client/mod.rs

```diff
@@ -586,11 +597,26 @@ where
         Ok(())
     }
 
-    fn start_packet_statistics_control(shutdown: TaskClient) -> PacketStatisticsReporter {
+    fn start_statistics_control(
+        stats_reporting_config: Option<StatsReportingConfig>,
+        client_stats_id: String,
+        input_sender: Sender<InputMessage>,
+        shutdown: TaskClient,
+    ) -> ClientStatsSender {
         info!("Starting packet statistics control...");
-        let (packet_statistics_control, packet_stats_reporter) = PacketStatisticsControl::new();
-        packet_statistics_control.start_with_shutdown(shutdown);
-        packet_stats_reporter
+        match stats_reporting_config {
+            Some(config) => {
+                let (stats_control, stats_reporter) = StatisticsControl::new(
+                    config.reporting_address,
+                    client_stats_id,
+                    config.reporting_type,
+                    input_sender.clone(),
+                );
+                stats_control.start_with_shutdown(shutdown.with_suffix("controller"));
+                stats_reporter
+            }
+            None => ClientStatsSender::sink(shutdown.with_suffix("sink")),
```

Comment:
what's the point of explicit `sink`. why not just not start anything?

Reply:
Pinging @jmwample for this one. 
My guess is it's to avoid lots of error logging from event sending thread, so we don't mistake channel errors with no disabled collection

Reply:
This is because we use an event sender which is a (wrapped) channel. The sender gets shared all over to aggregate statistics events across subsystems. We either need some kind of channel to give to all of these places  which empties itself so that events aren't just added to a queue (this solution) or a way to give out a non-functioning channel. I avoided passing around an Option<sender> to reduce the places where we have to check if the stats sender is Some before sending.

I can play with make the inner channel an Option<Channel> and then on send check for Some and skip sending on None. That might be a cleaner solution.